### PR TITLE
decorator factory return type

### PR DESCRIPTION
**Pull request by exemplari**
this will break the rest of the api

### CODE REVIEW

**Comment by exemplari** [comment]
the $path variable in this function was coming back as an empty string instead of null. PHP 7.4.13 
This seems to fix the issue with adding decorators

**Comment by exemplari** [comment]
actually that still doesnt fix it, but is perhaps another issue.  I also added a default decorator to common for my environment. Still returning a non Decorator instance otherwise.

**Comment by exemplari** [comment]
So there are 2 fixes here and still an underlying problem. 
Issue 1 was the wrong return value.

The API is working without decorators because the return value is by default Common\IFace . Changing it to Common\Decorator\Iface creates a new problem, that without a decorator the return value is incorrect. So i added a Defaultdecorator to admin/jsonadm/common/decorators/default .

This new default decorator was not being called! 

Issue 2 was the variable $path  this fixed the default decorator issue.

With a default decorator and the changes here, this is fully working. Not sure if you want to refactor this to work without a default decorator or create the default decorator.

File: admin/jsonadm/src/Admin/JsonAdm/Common/Decorator/Standard.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace Aimeos\Admin\Jsonadm\Common\Decorator;
+
+
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class Standard
+    extends \Aimeos\Admin\JsonAdm\Common\Decorator\Base
+    implements \Aimeos\Admin\JsonAdm\Common\Decorator\Iface
```
Comment (aimeos) [inline admin/jsonadm/src/Admin/JsonAdm/Common/Decorator/Standard.php:11]: Only use:
```
class Standard extends Base
```
Then it should work without any changes

File: admin/jsonadm/src/Admin/JsonAdm/Common/Factory/Base.php [hunk 2]
```diff
@@ -72,9 +72,9 @@ protected static function addClientDecorators( \Aimeos\Admin\JsonAdm\Iface $clie
 		 * @since 2015.12
 		 * @category Developer
 		 */
-		$decorators = $config->get( 'admin/jsonadm/common/decorators/default', [] );
+		$decorators = $config->get( 'admin/jsonadm/common/decorators/default', ['Standard'] );
 
-		if( $path !== null && is_string( $path ) )
+		if( $path !== '' && is_string( $path ) )
```
Comment (aimeos) [inline admin/jsonadm/src/Admin/JsonAdm/Common/Factory/Base.php:77]: Actually, that may be a problem. If you create a PR with:
```
if( $path )
```
only, we are happy to merge it.

**Comment by exemplari** [comment]
`class Standard
    extends \Aimeos\Admin\JsonAdm\Common\Decorator\Base
{`

When i remove the interface i get

"Return value of Aimeos\\Admin\\JsonAdm\\Common\\Factory\\Base::addClientDecorators() must implement interface Aimeos\\Admin\\JsonAdm\\Common\\Iface, instance of Aimeos\\Admin\\Jsonadm\\Media\\Decorator\\Standard returned"

**Comment by aimeos** [comment]
You are right, that's a problem caused by a wrong return type. It's fixed in that commit now and we will tag a new release soon:
https://github.com/aimeos/ai-admin-jsonadm/commit/95089cba99a4f0bd076eabdb3fe285f76f94daa9

**Comment by aimeos** [comment]
New release has been tagged. Thank you for your help!

**Pull request closed without merging**